decoder/src/test/java/openlr/decoder/data/OpenLRRatingImplTest.java

```diff
@@ -236,6 +255,84 @@ public final void testRating3() {
         }
     }
 
+    /**
+     * Rating test of junction node when applying no non-junction node factor
+     */
+    @Test
+    public final void testRatingJunctionNodeNoNonJunctionNodeFactor() {
```

Comment:
It may improve the readability of these tests to make them a parametrized test. Then a read could skim over the data "table" and may grasp quickly what the actual difference is between the cases.